lib/core/extension.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:cake_wallet/core/auth_service.dart';
+import 'package:cake_wallet/di.dart';
+import 'package:cake_wallet/routes.dart';
+import 'package:cake_wallet/src/screens/auth/auth_page.dart';
+import 'package:flutter/material.dart';
+
+const List<String> _alwaysAuthenticateRoutes = [
+  Routes.showKeys,
+  Routes.backup,
+  Routes.setupPin,
+];
+
+final _authService = getIt.get<AuthService>();
+
+extension AuthenticateRoute on BuildContext {
+  void navigateToAuthenticatedRoute(
+      {Function(bool)? onAuthSuccess, String route = '', Object? arguments}) {
```

Comment:
**Minor:**
`route` can be nullable and add an assertion instead to make sure either `onAuthSuccess` or `route`  is passed

Reply:
since it's not only navigation, we can use a more generic name
like `authenticateAction` for example